docs/v1/P4Runtime-Spec.mdk

```diff
@@ -4385,29 +4385,30 @@ them, and/or process updates in parallel.  Updates across multiple concurrent
 `WriteRequest`s can also be processed interleaved and/or in parallel.
 However, **the processing of updates must be strictly serializable**.  That
 is, given a history $S$ of `WriteRequest`s including the responses to those
-requests, there must exist an order $L$ for all updates in $S$, such that:
+requests, there must exist an order $L$ for all requests in $S$, such that:
```

Comment:
By changing this to not talk about updates anymore, we lose the formal definition about what kind of reordering of updates are allowed. I think we should keep this at the update level and just add the necessary restriction.

Reply:
That makes sense, even though I feel the definition becomes a bit too complicated compared to something like this meant for databases: http://www.bailis.org/blog/linearizability-versus-serializability/. I guess it makes sense in a way since DB transactions are usually assumed to execute operations in order, whereas we want updated within a request to be able to be processed in parallel.

Reply:
I did change "the processing of updates must be strictly serializable" to "the processing of requests must be strictly serializable" though, as the first one is not accurate IMO.

Reply:
My mental P4runtime API specification glossary cache is fairly empty right now.  Looking to refresh it here :-)

Is the intent within the P4Runtime spec is that "update" refers to a single insert/update/delete operation, whereas request refers to an entire `WriteRequest` message?

Reply:
OK, my mental cache is getting filled again.  A request is a write request or read request, and write requests contain unordered sets of updates.  Still reading and thinking on these changes, but should be done soon.

Reply:
this section mostly concerns itself with write requests, but I guess the statement "the processing of requests must be strictly serializable" holds even if requests == writes + reads.

yes update refers to an individual write operation in a write request batch, the name comes from the field name in p4runtime.proto

Reply:
Sorry for the delay, I was sick last week.

Looks good, thanks for the updates. One last comment: Right now we say "there must exist an order $L$ for all requests in $S$", but I think $L$ is now again an order over updates, right? Should we revert that back to updates, or maybe "all updates of all requests"?

Reply:
Yes, I definitely forgot to revert that one